core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3877,13 +3883,41 @@ public static double lesser(double b0, double b1) {
     return b0 > b1 ? b1 : b0;
   }
 
+  public static @Nullable <T extends Comparable<T>> List<T> lesser(
+      @Nullable List<T> b0, @Nullable List<T> b1) {
+    if (b0 == null) {
+      return b1;
+    }
+    if (b1 == null) {
+      return b0;
+    }
+    return lt(b0, b1) ? b0 : b1;
+  }
+
+  public static @Nullable <T extends Comparable<T>> List<T> greater(
```

Comment:
Add JavaDoc

Reply:
there are at least 10 versions of this function in this file, which share one common comment.